sdk/keyvault/keyvault-keys/test/public/crypto.hsm.spec.ts

```diff
@@ -1,108 +1,102 @@
-        cryptoClient = new CryptographyClient(
-          keyVaultKey.id!,
-          credential,
-          recorder.configureClientOptions({ disableChallengeResourceVerification: !isLiveMode() }),
-        );
-        const text = this.test!.title;
-        const encryptResult = await cryptoClient.encrypt({
-          algorithm: "A256GCM",
-          plaintext: stringToUint8Array(text),
-        });
-        assert.exists(encryptResult.iv);
-        assert.exists(encryptResult.authenticationTag);
+  describe("with AES crypto algorithms", async function () {
+    it("encrypts and decrypts using AES-GCM", async function (ctx) {
+      keyVaultKey = await hsmClient.createKey(keyName, "AES", { keySize: 256 });
+      cryptoClient = new CryptographyClient(
+        keyVaultKey.id!,
+        credential,
+        recorder.configureClientOptions({ disableChallengeResourceVerification: !isLiveMode() }),
+      );
+      const text = ctx.task.name;
+      const encryptResult = await cryptoClient.encrypt({
+        algorithm: "A256GCM",
+        plaintext: stringToUint8Array(text),
+      });
+      assert.exists(encryptResult.iv);
+      assert.exists(encryptResult.authenticationTag);
 
-        const decryptResult = await cryptoClient.decrypt({
-          algorithm: "A256GCM",
-          ciphertext: encryptResult.result!,
-          iv: encryptResult.iv!,
-          authenticationTag: encryptResult.authenticationTag!,
-        });
-        assert.equal(text, uint8ArrayToString(decryptResult.result));
-        await testClient?.flushKey(keyName);
+      const decryptResult = await cryptoClient.decrypt({
+        algorithm: "A256GCM",
+        ciphertext: encryptResult.result!,
+        iv: encryptResult.iv!,
+        authenticationTag: encryptResult.authenticationTag!,
       });
+      assert.equal(text, uint8ArrayToString(decryptResult.result));
+      await testClient?.flushKey(keyName);
+    });
 
-      it("encrypts and decrypts using AES-CBC", async function (this: Context) {
-        if (!isNode) {
-          this.skip();
-        }
-        keyVaultKey = await hsmClient.createKey(keyName, "AES", { keySize: 256 });
-        cryptoClient = new CryptographyClient(
-          keyVaultKey.id!,
-          credential,
-          recorder.configureClientOptions({ disableChallengeResourceVerification: !isLiveMode() }),
-        );
-        const text = this.test!.title;
-        // We are using a predictable IV to support our recorded tests; however, you should use a cryptographically secure IV or omit it and
-        // let the client library generate it for you.
-        const iv = stringToUint8Array("xxxxxxxxxxxxxxxx");
-        const encryptResult = await cryptoClient.encrypt({
-          algorithm: "A256CBCPAD",
-          plaintext: stringToUint8Array(text),
-          iv,
-        });
+    it("encrypts and decrypts using AES-CBC", async function (ctx) {
+      if (!isNode) {
```

Comment:
nitpick: Both `isNode` and `isNodeLike` are used in the tests, is that necessary? also, could the node specific tests be in a node folder and the browser specific tests be in a browser folder? This should improve readability and maintainability and avoids funny bundling issues.

Reply:
Yea, this is a great callout. Honestly with browser tests skipped maybe I can just remove all the `isNode` checks entirely?